x-pack/plugins/infra/public/pages/metrics/metric_detail/index.tsx

```diff
@@ -19,8 +19,6 @@ export const MetricDetail = () => {
     params: { type: nodeType, node: nodeName },
   } = useRouteMatch<{ type: InventoryItemType; node: string }>();
 
-  const PageContent = () => (nodeType === 'host' ? <AssetDetailPage /> : <MetricDetailPage />);
```

Comment:
This was the cause of the search inputs losing focus. This was creating a new `PageContent` on every render and forced react to re-create the DOM nodes.

Reply:
ah, well. that makes the PR I just opened useless then: https://github.com/elastic/kibana/pull/165288 :)